include/swift/Sema/ConstraintSystem.h

```diff
@@ -4436,7 +4441,10 @@ class ConstraintSystem {
       llvm::function_ref<bool(const ClosureExpr *)> isolatedByPreconcurrency =
         [](const ClosureExpr *closure) {
           return closure->isIsolatedByPreconcurrency();
-        });
+        },
+      llvm::function_ref<bool(Expr *)> isAssignTarget = [](Expr *) {
+        return true;
```

Comment:
Is this the right default?

Reply:
It's the default that keeps existing code working the same way as it has before. I could remove the default and force all callers to choose.

Reply:
Maybe this is going to be a non-issue if you go with walk approach since it could just be a method on a constraint system?

Reply:
I've cleaned this up and removed the default, thanks for the nudge.